src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
I tend to agree with @traversaro also because of my mindset :smile:: I always go for basic standard constructs when possible rather than using fancy solutions, unless they bring solid improvements, but here two `if`'s among integers cost practically `0`.

However, if we wanna stick to `try`+`catch` I think it'd be worth arranging the code in several lines to improve readability and/or perhaps resort to macros.